lib/palindrome_check.rb

```diff
@@ -1,5 +1,34 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  return false if my_phrase.nil?
+
+  current_index = 0
+  back_index = my_phrase.length - 1
+  mismatch = 0
+
+  while current_index < my_phrase.length
+
+    if my_phrase[current_index] == " "
```

Comment:
To account for:
1. multiple, consecutive white spaces
2. white spaces at both current_index as well as back_index
Consider: 
- not having the back_index moving forward not as an elsif but a separate condition.
- converting the white space checking into a loop instead of conditional statements.